### DISCUSSION TITLE
模板编译问题

### DISCUSSION THREAD

**Discussion by LiY321** [sjtug/SJTUThesis#699] [提问] [answered]
我在window系统下使用compile.bat 编译，显示Error! Please check the 'main.log' for more details . . .
用MiKTeX进行编译，也会报错 Undefined control sequence. l.4 \documentclass [type=bachelor]{sjtuthesis}

**Comment by AlexaraWu** [comment, marked as answer]
MiKTeX 使用脚本 `compile.bat` 编译需要自行安装一个 [perl](https://strawberryperl.com/) 解释器。

**Reply by LiY321** [reply]
现在我使用compile.bat脚本能够编译了，但是我使用TeXworks作为ide编译的时候，仍然会报错
![Snipaste_2021-06-29_02-12-52](https://user-images.githubusercontent.com/65121771/123684056-90fedc80-d87f-11eb-9e6e-70d0ec74e8fd.png)
![Snipaste_2021-06-29_02-13-04](https://user-images.githubusercontent.com/65121771/123684059-92300980-d87f-11eb-80d2-814c98909bc2.png)

**Reply by skyzh** [reply]
是否使用了 latexmk 作为 TeXworks 的编译器？

**Reply by LiY321** [reply]
怎么设置latexmk作为编译器呢？

**Comment by LiY321** [comment]
我用texlive和texstudio编译，我的配置是这样的
![微信截图_20210829003724](https://user-images.githubusercontent.com/65121771/131225273-4f130452-ed78-44ff-ba13-04f221860c75.png)
然后错误信息是这样的
![微信截图_20210829003736](https://user-images.githubusercontent.com/65121771/131225310-a989e1d8-9553-4a66-b40d-1601ce11417c.png)
![微信截图_20210829003844](https://user-images.githubusercontent.com/65121771/131225311-aa21a649-b838-45f4-8b19-4faed6b5dde0.png)

**Reply by skyzh** [reply]
需要 XeLaTeX。

**Reply by skyzh** [reply]
为啥不用 compile.bat？

**Reply by LiY321** [reply]
因为可以反向定位，预览也方便一点。可是我找了很久也不知道怎么把engine改成XeLaTeX。如果我把编译器改为XeLaTeX，就会找不到sjtuthesis.cls

**Reply by skyzh** [reply]
VSCode 也支持反向定位。TeXStudio 应该没有开发者在用，如果你试成功了可以更新一下我们的 readme。

**Reply by skyzh** [reply]
看起来是 latexmk 的参数问题，你的 log 里显示调用了 `latexmk -pdf`。理论上应该 `latexmk -xelatex`。

**Reply by LiY321** [reply]
成功了，感谢。确实是应该把latexmk的命令参数改成 -xelatex

**Comment by hushidong** [comment]
默认文献工具改成biber hzzmail ***@***.*** 签名由 网易邮箱大师 定制 在2021年08月29日 01:19，Alex Chi 写道： 为啥不用 compile.bat？ — You are receiving this because you are subscribed to this thread. Reply to this email directly, view it on GitHub, or unsubscribe. Triage notifications on the go with GitHub Mobile for iOS or Android.
